components/suggest/src/db.rs

```diff
@@ -124,16 +125,28 @@ impl<'a> SuggestDao<'a> {
 
     /// Fetches suggestions that match the given keyword from the database.
     pub fn fetch_by_keyword(&self, keyword: &str) -> Result<Vec<Suggestion>> {
-        self.conn.query_rows_and_then_cached(
-            "SELECT s.id, k.rank, s.title, s.url, s.provider
+        let (keyword_prefix, keyword_suffix) = match keyword.find(' ') {
+            Some(index) => {
+                let (prefix, suffix) = keyword.split_at(index);
+                (prefix.to_string(), suffix[1..].to_string())
+            }
+            None => (keyword.to_string(), String::new()),
+        };
+        Ok(self.conn.query_rows_and_then_cached(
+            "SELECT s.id, k.rank, s.title, s.url, s.provider, -1 as confidence_type
```

Comment:
I'd recommend using `NULL` instead of `-1` for the missing field. Also, how about we move this field to `pocket_custom_details` as it's Pocket-specific? We will need to do more in the handler, but I think it's still better than introducing a dummy column to the suggestion table.

WDYT?

Reply:
sooo confidence is sits in the pocket_keywords table, the -1 is used as a dummy value cause UNION requires the same number of columns. Is there a way to smush the two results together without having the same number of columns?

Reply:
Right, it's in the pocket_keywords table. If we don't include it here, we'd have to do another lookup against the keywords table. NVM, let's use NULL as the dummy value then.